### PR TITLE
containers okr - cluster agent refresh

### PR DESCRIPTION
**Pull request by cswatt**
### What does this PR do?

**arranging and consolidating information**
- very short and dubiously useful `containers/cluster_agent/build` moved to `containers/faq/build_cluster_agent`
- very short and dubiously useful `containers/cluster_agent/metadata_provider` deleted. it seems the steps to enable the metadata provider were identical to the steps to enable the cluster agent in general.
- very short `containers/cluster_agent/event_collection` content moved to `containers/kubernetes/configuration`
- relatively-esoteric `containers/cluster_agent/clusterchecksrunner` moved to a guide. linked to from cluster checks page.
- the "custom & external metrics" page is better contextualized as a guide to using HPA with custom & external metrics. it is now a guide that is linked to from the cluster agent index.
- some nav-reordering to better communicate the complexity/relevance of things

**pedantry**
- renamed "cluster checks runner" to "cluster check runner"
- renamed "endpoints check" to "endpoint check"
- capitalized Pod when used in the context of Kubernetes Pods
- general edit/rearrangement of content

### Motivation
okr

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/containers/cluster_agent/troubleshooting.md [hunk 1]
```diff
@@ -287,6 +290,214 @@ And indeed `824 * 3 replicas = 2472`.
...
+When leader election is enabled, only the leader serves Cluster Check configurations to the node-based Agents. If only one replica of the Cluster Agent Pod is running, it is the leader. Otherwise, you can identify the name of the leader in the `datadog-leader-election` `ConfigMap`:
+
+```yaml
+# kubectl get cm datadog-leader-election -o yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  annotations:
+    control-plane.alpha.kubernetes.io/leader: '{"holderIdentity":"cluster-agent-rhttz", ...''
+```
```
Comment (apigirl) [inline content/en/containers/cluster_agent/troubleshooting.md:306]: I think this code example is going wrong

Comment (apigirl) [reply content/en/containers/cluster_agent/troubleshooting.md:306]: It looks fine in the staging though so 🤷‍♀️ https://docs-staging.datadoghq.com/cswatt/cluster-agent-okr/containers/cluster_agent/troubleshooting/#autodiscovery-in-the-cluster-agent

File: content/en/containers/cluster_agent/admission_controller.md [hunk 2]
```diff
@@ -27,11 +24,11 @@ Datadog's admission controller is `MutatingAdmissionWebhook` type. For more deta
 ## Configuration
 {{< tabs >}}
 {{% tab "Helm chart" %}}
```
Comment (apigirl) [inline content/en/containers/cluster_agent/admission_controller.md:26]: I noticed when I was reviewing the preview that you name the tabs differently on this page than on other pages. Normally it's just `Helm` `Operator`

File: content/en/containers/faq/build_cluster_agent.md [hunk 3]
```diff
@@ -1,8 +1,9 @@
 ---
 title: Build the Datadog Cluster Agent
-kind: documentation
+kind: faq
```
Comment (apigirl) [inline content/en/containers/faq/build_cluster_agent.md:3]: Always surprised to see things move into the FAQ, so just want to check that it's very intentional

**Review by apigirl — APPROVED** [review]
just a few things that honestly probably aren't even issues

File: content/en/containers/faq/build_cluster_agent.md (hunk 3 above)
Comment (cswatt) [reply content/en/containers/faq/build_cluster_agent.md:3]: this one's intentional—info doesn't seem super useful to most people but i don't want to delete it (yet)

File: content/en/containers/cluster_agent/troubleshooting.md (hunk 1 above)
Comment (cswatt) [reply content/en/containers/cluster_agent/troubleshooting.md:306]: hmm i think it's rendering fine, but i'll clean up the end of that example string

**Pull request merged**